src/test/compile-fail/auxiliary/foo/bar.rs

```diff
@@ -0,0 +1,12 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+pub mod baz; //~ ERROR cannot declare a new module at this location
+             //~ NOTE maybe move this module `src/test/compile-fail/foo/bar` to its own directory via `src/test/compile-fail/foo/bar/mod.rs`
```

Comment:
if the suggestion is saying move `foo/bar` to `foo/bar/mod.rs`, wouldn't it make more sense for it to suggesting moving `foo/bar.rs` (i.e. including the file extension) to `foo/bar/mod.rs`, so that the source and destination look consistent?